### PR TITLE
[compiler service process] Support client function in assertions

### PR DESCRIPTION
**Pull request by miherlosev**
Changes:
* processes have clear titles (for debug purposes)
* ReExecutablePromise (ClientFunction and Selector results) executed into the service process
* typings and refactoring

### CODE REVIEW

**Comment by miherlosev** [comment]
Test are green

All tasks are green (except `travis.test-functional-local-headless-firefox`) https://github.com/DevExpress/testcafe/pull/6000#issuecomment-788878684
`travis.test-functional-local-headless-firefox` is green (https://travis-ci.com/github/testcafe-build-bot/testcafe/jobs/487450089)

File: @types/map-reverse/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,5 @@
+declare module 'map-reverse' {
```
[the remark this answers is not in the document]
Comment (AndreyBelym) [reply @types/map-reverse/index.d.ts:1]: BTW great thanks for constantly improving our type safety and porting more and more modules to Typescript each time 👍

**Comment by AndreyBelym** [comment]
Everything is perfect, but using this micromodule... Seems odd a bit since we have lodash

File: @types/map-reverse/index.d.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (miherlosev) [reply @types/map-reverse/index.d.ts:1]: Ok. I will remove the `map-reverse` module in the separate PR.

**Comment by miherlosev** [comment]
The tests are green.
All tests are green (except `travis.test-functional-local-headless-chrome`) https://github.com/DevExpress/testcafe/pull/6000#issuecomment-790606339
`travis.test-functional-local-headless-chrome` is green https://travis-ci.com/github/testcafe-build-bot/testcafe/jobs/488212538

**Pull request merged**